payments_service/braintree/tests/test_views.py

```diff
@@ -332,6 +332,89 @@ def test_user_already_subscribed(self):
         assert self.solitude.braintree.mozilla.subscription.get.called
 
 
+class TestChangeSubscriptionPayMethod(AuthenticatedTestCase):
+
+    def setUp(self):
+        super(TestChangeSubscriptionPayMethod, self).setUp()
+        self.url = reverse('braintree:subscriptions.paymethod.change')
+        self.new_pay_method_uri = '/new_pay_method_uri'
+        self.subscription_uri = '/subscription_uri'
+
+    def setup_subscription_post(self):
+        bt = self.solitude.braintree
+        bt.subscription.paymethod.change.post.return_value = {}
+        return bt.subscription.paymethod.change.post
+
+    def setup_uri_lookups(self, uri_404=None):
+        resources = {
+            self.subscription_uri: mock.Mock(),
+            self.new_pay_method_uri: mock.Mock(),
+        }
+
+        def get_api_object(uri):
+            res = resources.get(uri)
+            if not res:
+                raise ValueError('unknown uri {}'.format(uri))
+            if uri_404 == uri:
+                res.get.side_effect = HttpClientError('not found')
+            res.get.return_value = {}
+            return res
+
+        self.solitude.by_url.side_effect = get_api_object
+        return resources
+
+    def post(self):
+        return self.json(self.client.post(self.url, {
+            'new_pay_method_uri': self.new_pay_method_uri,
+            'subscription_uri': self.subscription_uri,
+        }))
+
+    def test_change_paymethod(self):
+        self.setup_uri_lookups()
+        api_post = self.setup_subscription_post()
+
+        res, data = self.post()
+        eq_(res.status_code, 204, res)
```

Comment:
Oh wait, ignore me, you are returning a 204 in the service so we don't share all that useful info up to the world.